### PR TITLE
Separate Navigation and Data contexts

### PR DESCRIPTION
**Pull request by pasanchMSFT**
### Summary of changes 🔍 
Separated `WizardNavigationContext` data management into its own context: `DataManagementContext`. `DataManagementContext` still only has simple actions and reducers from components are yet to be migrated into it, this will be discussed further with the team and potentially be done in other PRs. 

Components now take care of their own view models and commit to the global data store on unmount or navigation. This pattern can be seen in `DataSourceStep` component and `ModelLists` component. 

Note:
TwinLists component requires a lot of work, would be best to leave that to another PR.

Also in this PR:
- Minor styling changes to TwinVerificationStep, which will probably be reworked later.
- File structure modification, moved Contexts into a folder below Legion folder.
- i18n for Twin verification step components.

TODO (potentially in other PRs)
- Rename steps to match newer patterns
- UX of Modification step update
- Content pass

### Testing 🧪
Open any step stories and modify values.

### Checklist ✔️
- [ ] Linked associated issue (if present)
- [ ] Added relevant labels
- [ ] Requested developer reviews
- [ ] Request UI review from design / PM
- [ ] Verify all code checks are passing

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -39312,7 +39312,7 @@
             "dev": true,
             "requires": {
                 "exec-sh": "^0.3.2",
-                "minimist": "^1.2.0"
```
Comment (iremgokce) [inline package-lock.json:39315]: why is there a package_lock.json changes?

Comment (pasanchMSFT) [reply package-lock.json:39315]: I did npm install to install @msnyder-msft changes, and this came out

Comment (msnyder-msft) [reply package-lock.json:39315]: did you do `npm ci`?

Comment (pasanchMSFT) [reply package-lock.json]: resolved with npm ci --legacy-peer-deps

File: src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/TwinVerificationStep.types.ts [hunk 2]
```diff
@@ -26,19 +28,15 @@ export interface ITwinVerificationStepStyles {
...
 
 export interface ITwinVerificationStepSubComponentStyles {
-    button: Partial<IButtonStyles>;
+    button?: Partial<IButtonStyles>;
 }
 
-export interface IViewModelFromCooked {
-    id: string;
-    name: string;
+export interface IModelExtended extends IModel {
```
Comment (iremgokce) [inline src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/TwinVerificationStep.types.ts:34]: I think this was my original intention, yeah this is better.

File: src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+export interface IDataManagementContext {
+    dataManagementContextState: IDataManagementContextState;
+    dataManagementContextDispatch: React.Dispatch<DataManagementContextAction>;
+}
+
+export interface IDataManagementContextState {
+    // Keep adding source types here
+    sources: Array<IADXConnection | PIDDocument>;
+    initialAssets: IAppData;
```
Comment (iremgokce) [inline src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts:11]: not sure if we need to keep initialAssets and modifiedAssets separately but I think it is fine for now.

File: src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          };
+      }
+    | {
+          type: DataManagementContextActionType.SET_INITIAL_ASSETS;
+          payload: {
+              data: IAppData;
+          };
+      }
+    | {
+          type: DataManagementContextActionType.UPDATE_MODIFIED_ASSETS;
```
Comment (iremgokce) [inline src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts:38]: for consistency:
```suggestion
          type: DataManagementContextActionType.SET_MODIFIED_ASSETS;
```

File: src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx [hunk 5]
```diff
@@ -1,127 +1,172 @@
...
 
     //callbacks
     const handleOnCheck = useCallback(
         (modelId: string, propertyId: string, checked: boolean) => {
-            wizardNavigationContextDispatch({
-                type:
-                    WizardNavigationContextActionType.SET_MODEL_PROPERTY_SELECTED,
-                payload: { modelId, propertyId, checked }
-            });
+            const tempModels = Array.from(models);
```
Comment (iremgokce) [inline src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx:82]: use `deepCopy`
```suggestion
            const tempModels = deepCopy(models);
```

File: src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx [hunk 6]
```diff
@@ -1,127 +1,172 @@
...
+            if (modelToModifyIdx >= 0) {
+                if (checked) {
+                    tempModels[modelToModifyIdx].propertyIds.push(propertyId);
+                } else {
+                    tempModels[modelToModifyIdx].propertyIds.filter(
+                        (p) => p !== propertyId
+                    );
+                }
+            }
+            setModels(tempModels);
```
Comment (iremgokce) [inline src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx:95]: So here shouldn't we also update the modifiedAssets in DataManagementContext? or is it done in "next/finish" button in TwinVerificationStep - I didn't see that there either?

File: src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/TwinVerificationStep.tsx [hunk 7]
```diff
@@ -77,40 +80,47 @@ const TwinVerificationStep: React.FC<ITwinVerificationStepProps> = (props) => {
...
-                        <TwinLists />
-                        <PrimaryButton
-                            text="Next"
-                            styles={classNames.subComponentStyles.button()}
-                            onClick={onFinishButtonClick}
-                        />
-                    </PivotItem>
-                </Pivot>
-            </div>
+            <Stack>
```
Comment (msnyder-msft) [inline src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/TwinVerificationStep.tsx:83]: nit: we can just put the root style on the stack itself here.

File: src/Apps/Legion/Components/WizardShell/WizardShellMockData.ts [hunk 8]
```diff
@@ -109,29 +105,24 @@ export const steps: IStepperWizardStep[] = [
     }
 ];
 
-export const cookedData: ICookAssets = {
+export const appData: IAppData = {
```
Comment (msnyder-msft) [inline src/Apps/Legion/Components/WizardShell/WizardShellMockData.ts:108]: let's reanmae this to `DEFAULT_MOCK_WIZARD_SHELL_DATA`

File: src/Apps/Legion/Components/WizardShell/WizardShellMockData.ts [hunk 9]
```diff
@@ -109,29 +105,24 @@ export const steps: IStepperWizardStep[] = [
...
-        models: getViewModelsFromCookedAssets(mockModels),
-        twins: getViewTwinsFromCookedAssets(
-            mockTwins,
-            getViewModelsFromCookedAssets(mockModels)
-        ),
-        properties: mockProperties
-    },
-    relationshipStepData: null,
-    finishStepData: null
+export const wizardData: IDataManagementContextState = {
```
Comment (msnyder-msft) [inline src/Apps/Legion/Components/WizardShell/WizardShellMockData.ts:116]: let's rename this to `DEFAULT_MOCK_DATA_MANAGEMENT_STATE`

File: src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts (hunk 3 above)
Comment (msnyder-msft) [reply src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts:11]: we only need them right now to be able to compare and know what was changed during this session. So if you come back from a future step we can build out the view model correctly again, like if you unselect a property we know whether to check it. There are other ways to do track that, but this seemed like a decent naive implementation that would extend to other scenarios.

File: src/Apps/Legion/Models/Interfaces.ts [hunk 10]
```diff
@@ -44,15 +44,25 @@ export interface IRelationship {
...
+/** to be used for Database ingestion flow */
+export interface IADXConnection {
+    selectedSourceDatabase: string;
+    selectedSourceTable: string;
+    selectedSourceTwinIDColumn: string;
+    selectedSourceTableType: string;
+    selectedTargetDatabase: string;
+}
+
+export interface PIDDocument {
```
Comment (msnyder-msft) [inline src/Apps/Legion/Models/Interfaces.ts:65]: nit: prefix with I

**Review by msnyder-msft — APPROVED** [review]
Good enough for now. Left some naming stuff and nits.

**Comment by msnyder-msft** [comment]
Looks like hte graph step was broken. Can you wire up the data?

File: src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts (hunk 3 above)
Comment (pasanchMSFT) [reply src/Apps/Legion/Contexts/DataManagementContext/DataManagementContext.types.ts:11]: we need this to track initial values and selected/unselected values.

File: src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx [hunk 11]
```diff
@@ -1,127 +1,172 @@
...
+    const handleModelNameChange = useCallback(
+        (modelId: string, modelName: string) => {
+            const tempModels = Array.from(models);
+            const modelToModifyIdx = tempModels.findIndex(
+                (m) => m.id === modelId
+            );
+            if (modelToModifyIdx >= 0) {
+                tempModels[modelToModifyIdx].name = modelName;
+            }
+            setModels(tempModels);
```
Comment (iremgokce) [inline src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx:109]: same here: So here shouldn't we also update the modifiedAssets in DataManagementContext? or is it done in "next/finish" button in TwinVerificationStep - I didn't see that there either?

**Review by iremgokce — APPROVED** [review]
Left some comments, nothing major.

File: src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx (hunk 6 above)
Comment (pasanchMSFT) [reply src/Apps/Legion/Components/WizardShell/Internal/TwinVerificationStep/Internal/ModelLists.tsx:95]: that gets called on unmount in the useEffect above this

File: src/Apps/Legion/Components/WizardShell/Internal/DataSourceStep/DataSourceStep.tsx [hunk 12]
```diff
@@ -73,10 +71,11 @@ const DataSourceStep: React.FC<IDataSourceStepProps> = (props) => {
     );
 
     const [adapterResult, setAdapterResult] = useState(null);
-    const [cookAssets, setCookAssets] = useState<ICookAssets>(null);
+    const [appData, setAppData] = useState<IAppData>(null);
```
[the remark this answers is not in the document]
Comment (pasanchMSFT) [reply src/Apps/Legion/Components/WizardShell/Internal/DataSourceStep/DataSourceStep.tsx:74]: good point, I think we should rework this, but don't want to block on it.

File: src/Apps/Legion/Components/WizardShell/WizardShellMockData.ts (hunk 8 above)
Comment (pasanchMSFT) [reply src/Apps/Legion/Components/WizardShell/WizardShellMockData.ts:108]: this is internal so I'll remove the export and rename to mockAppData

**Pull request merged**